src/core/config/key_values.rs

```diff
@@ -3,27 +3,40 @@ use std::ops::Deref;
 
 use serde::{Deserialize, Deserializer, Serialize, Serializer};
 
+use crate::core::is_default;
+
+type Value = (String, Option<bool>);
+
 #[derive(Clone, Debug, Default, Eq, PartialEq, schemars::JsonSchema)]
-pub struct KeyValues(pub BTreeMap<String, String>);
+pub struct KeyValues(pub BTreeMap<String, Value>);
 
 impl Deref for KeyValues {
-    type Target = BTreeMap<String, String>;
+    type Target = BTreeMap<String, Value>;
 
     fn deref(&self) -> &Self::Target {
         &self.0
     }
 }
 
-impl FromIterator<(String, String)> for KeyValues {
-    fn from_iter<T: IntoIterator<Item = (String, String)>>(iter: T) -> Self {
+impl FromIterator<(String, Value)> for KeyValues {
+    fn from_iter<T: IntoIterator<Item = (String, Value)>>(iter: T) -> Self {
         KeyValues(BTreeMap::from_iter(iter))
     }
 }
 
 #[derive(Serialize, Deserialize, Clone, Debug, Default, Eq, PartialEq, schemars::JsonSchema)]
+#[serde(rename_all = "camelCase")]
 pub struct KeyValue {
     pub key: String,
     pub value: String,
+    #[serde(default, skip_serializing_if = "is_default")]
+    /// Determines whether to ignore query parameters with empty values when
+    /// forming URLs.
+    ///
+    /// When set to true, query parameters without values are completely ignored
+    /// during URL formation. When false (default), parameters without values
+    /// are included in the URL.
+    pub skip_empty: Option<bool>,
```

Comment:
We should keep the original `KeyValue` as it is because its being used in multiple places. Create a new one calle `URLQuery` type that has `key`, `value` and `skip_empty`.